pkg/actions/karpenter/create.go

```diff
@@ -73,6 +73,11 @@ func (i *Installer) Create() error {
 		return fmt.Errorf("failed to save the identity config: %w", err)
 	}
 
+	instanceProfileName := fmt.Sprintf("eksctl-%s-%s", builder.KarpenterNodeInstanceProfile, i.Config.Metadata.Name)
```

Comment:
Almost there. :) But, this won't work, because it's not what it creates. This is what you are looking for:

```go
	instanceProfileName := gfnt.MakeFnSubString(fmt.Sprintf("eksctl-%s-%s", KarpenterNodeInstanceProfile, k.clusterSpec.Metadata.Name))
	instanceProfile := gfniam.InstanceProfile{
		InstanceProfileName: instanceProfileName,
		Path:                gfnt.NewString("/"),
		Roles:               gfnt.NewSlice(roleRef),
	}
	k.newResource(KarpenterNodeInstanceProfile, &instanceProfile)
```

In the file `pkg/cfn/builder/karpenter.go`. This where the role is actually created. Here is where you should use this name or the generated name.

That's this part in this file:

```go
	// Create IAM roles
	taskTree := newTasksToInstallKarpenterIAMRoles(i.Config, i.StackManager, i.CTL.Provider.EC2())
	if err := doTasks(taskTree); err != nil {
		return err
	}
```

The config is passed in there too, so you can use that.

Reply:
My intent here was not to rename what eksctl creates but to allow you to use your own IAM instance profile, hence the nature of the changes. If I interpret correctly what you propose would change the name of the IAM instance profile created by eksctl in CFN correct?